spec/lib/git_spec.rb

```diff
@@ -22,7 +22,40 @@ module CodeClimate::TestReporter
         expect(Git).to receive(:`).once.with "git --git-dir=\"#{path}/.git\" help"
 
         Git.send :git, 'help'
+      end
+
+      context 'ensure logic that replies on Rails is robust in non-rails environments' do
+        before :all do
+          module ::Rails; end
+        end
+
+        after :all do
+          Object.send(:remove_const, :Rails)
         end
+
+        after :each do
+          Git.send :git, 'help'
+        end
+
+        it 'will check if constant Rails is defined' do
+          expect(Git).to receive(:configured_git_dir).once.and_return(nil)
+        end
+
+        it 'will not rails root if constant Rails is defined but does not respond to root' do
+          expect(Git).to receive(:configured_git_dir).once.and_return(nil)
+          expect(Rails).to receive(:root).twice.and_return('/path')
```

Comment:
In the implementation Rails.root would be called a third time if `Rails` responds (1st) and if `Rails.root.nil?` (2nd). I clarified this in a follow up commit.

```ruby
def rails_git_dir_present?
  const_defined?(:Rails) && Rails.respond_to?(:root) && !Rails.root.nil? &&
            File.directory?(File.expand_path('.git', Rails.root))
end
```